NIFI-1661 add random() function to expression language.

ckmcd opened the pull request.

jskora commented:
This looks pretty good, but in testing I came across 2 problems that need to be addressed before it can be merged.

1. The example in the [expression language guide](https://github.com/apache/nifi/pull/322/files#diff-daac74ec3b89e26d99806d7a90254fe3R1313) should be `${random():mod(10):plus(1)}` since EL uses [plus](http://nifi.apache.org/docs/nifi-docs/html/expression-language-guide.html#plus) for addition.

1. Though Math.Random appears to only return positive floats, `nextLong()` must be returning positive and negative numbers.  Using the example expression I get numbers between -100 and 100 showing in the attributes.  The documentation should reflect that this could be positive or negative and the example needs to be adjusted for that as well.

In spite of the problems, nice job incorporating the EL Guide changes!

ckmcd commented:
I've update the PR to address the issues (1 & 2) from above.  

1. Updated the EL Guide to use plus() instead of add()
2. Use Math.abs() so as only to generate non-negative numbers.  Updated EL guide to reflect this change.

jskora commented:
@ckmcd this has been merged to the 1.0 and 0.x branches.  Looks great and works well.

I forgot to put  "this closes" on the commit, could you please close the PR?
